eduiddashboard/verifications.py

```diff
@@ -109,69 +124,94 @@ def verify_nin(request, user, new_nin, reference=None):
-            old_user.retrieve_modified_ts(request.db.profiles)
-            old_user.save(request)
-            log.info('Removed mobile number from user {!r}.'.format(old_user))
+        old_user = DashboardUser(data=old_user_doc)
+        retrieve_modified_ts(old_user, request.dashboard_userdb)
+        if old_user and old_user.user_id != user.user_id:
+            log.debug('Found old user {!r} with phone number ({!s}) already verified.'.format(old_user, new_mobile))
+            log.debug('Old user phone numbers BEFORE: {!r}.'.format(old_user.phone_numbers.to_list()))
+            if old_user.phone_numbers.primary.key == new_mobile:
+                old_numbers = old_user.phone_numbers.to_list()
+                for number in old_numbers:
+                    if number.key != new_mobile:
+                        old_user.phone_numbers.primary = number.key
+                        break
+                else:
+                    old_user._phone_numbers = PhoneNumberList([])
+            old_user.phone_numbers.remove(new_mobile)
+            log.debug('Old user phone numbers AFTER: {!r}.'.format(old_user.phone_numbers.to_list()))
+            request.context.save_dashboard_user(old_user)
+            log.info('Removed phone number from user {!r}.'.format(old_user))
             steal_count += 1
     # Add the verified mobile number to the requesting user
-    user.add_verified_mobile(new_mobile)
-    log.info('Mobile number verified for user {!r}.'.format(user))
+    new_mobile_obj = PhoneNumber(data={'number': new_mobile,
+                                       'verified': True,
+                                       'primary': False})
+    log.debug('User had phones BEFORE verification: {!r}'.format(user.phone_numbers.to_list()))
+    if user.phone_numbers.primary is None:
+        log.debug('Setting NEW phone number to primary: {}.'.format(new_mobile_obj))
+        new_mobile_obj.is_primary = True
+    try:
+        user.phone_numbers.add(new_mobile_obj)
+    except DuplicateElementViolation:
+        user.phone_numbers.find(new_mobile).is_verified = True
+    log.info('Phone number verified for user {!r}.'.format(user))
     request.stats.count('dashboard/verify_mobile_stolen', steal_count)
     request.stats.count('dashboard/verify_mobile_completed', 1)
-    return user, _('Mobile {obj} verified')
+    return user, _('Phone {obj} verified')
 
 
 def verify_mail(request, user, new_mail):
-    user = OldUser(user)
     log.info('Trying to verify mail address for user {!r}.'.format(user))
     log.debug('Mail address: {!s}.'.format(new_mail))
     # Start by removing mail address from any other user
-    old_user_docs = request.userdb.get_users_by_filter({
-        'mailAliases': {'$elemMatch': {'email': new_mail, 'verified': True}}
-    }, raise_on_missing=False)
+    old_user_docs = request.dashboard_userdb._coll.find(
+            {'mailAliases': {'$elemMatch': {'email': new_mail, 'verified': True}}})
```

Comment:
doesn't this code (and other similar code) need to use a userdb.getuserby* function, and doesn't that function need to work with both old and new data in the am database simultaneously?